### ISSUE TITLE
Use a build pattern for types building

### ISSUE THREAD

**Issue by machacekondra** [imjoey/ovirt-engine-sdk-go#3]
Try to use the `builder` pattern also for the types, for the same
reasons. For example, the way to create an instance of the VM type
should be something this:

```go
  vm := NewVmBuilder().
    Name("myvm").
    Memory(1234).
    Cluster(
      NewClusterBuilder().
      Name("mycluser").
      Build()
    ).
    Template(
      NewTemplateBuilder().
      Name("mytemplate").
      Build()
    ).
    Build()
```

The use of builders also allows potential future improvements, like
adding validations inside the Build method.

**Comment by imjoey** [comment]
Fix it in #9 . New*Type*Builder() returns *Type* and error, not just only *Type*.

**Comment by imjoey** [comment]
Close it.
